File: java_solutions/179.largest-number.java
/*
 * @lc app=leetcode id=179 lang=java
 *
 * [179] Largest Number
 */

class Solution {
    public String largestNumber(int[] nums) {
        return Arrays.stream(nums) .mapToObj(String::valueOf) .sorted((a, b) -> (b + a).compareTo(a + b)) .collect(Collectors.collectingAndThen( Collectors.joining(), s -> s.startsWith("0") ? "0" : s));
    }
}

File: kotlin_solutions/179.largest-number.kt
/*
 * @lc app=leetcode id=179 lang=kotlin
 *
 * [179] Largest Number
 */

class Solution { fun largestNumber(nums: IntArray): String = nums.map { it.toString() } .sortedWith { a, b -> (b + a).compareTo(a + b) } .joinToString("") .let { if (it.all { c -> c == '0' }) "0" else it } }

